sydent/replication/peer.py

```diff
@@ -115,19 +111,37 @@ def verifyMessage(self, jsonMessage):
             e.requiredServername = self.servername
             raise e
 
+        # Get verify key from signing key
+        signing_key = signedjson.key.decode_signing_key_base64(alg, "0", self.pubkeys[alg])
+        verify_key = signing_key.verify_key
+
+        # Attach metadata
+        verify_key.alg = alg
+        verify_key.version = 0
+
         # Verify the JSON
         signedjson.sign.verify_signed_json(jsonMessage, self.servername, self.verify_key)
 
-    def pushUpdates(self, sgAssocs):
+    def pushUpdates(self, data):
+        """Push updates to a remote peer.
+
+        :param data: A dictionary of possible `sg_assocs`, `invite_tokens`
+            and `ephemeral_public_keys` keys.
+        :type data: Dict
+        :returns a deferred.
+        :rtype: Deferred
+        """
+
+        # sgAssocs is comprised of tuples (sgAssoc, shadowSgAssoc)
         if self.shadow:
-            body = {'sgAssocs': { k: v[1] for k, v in sgAssocs.items()}}
+            data["sg_assocs"] = { k: v[1] for k, v in data["sg_assocs"].items() }
```

Comment:
I wonder why we're putting both sgAssocs and shadowSgAssocs into a tuple only to pull them out again when we push them. Surely this can be done in a cleaner way. I'm going to try removing the middle steps.